lib/adapters/http/read.js

```diff
@@ -1,13 +1,94 @@
+'use strict';
 var _ = require('underscore');
 var contentType = require('content-type');
-var Juttle = require('../../runtime').Juttle;
+var AdapterRead = require('../../runtime/adapter-read');
 var parsers = require('../parsers');
+var errors = require('../../errors');
 var request = require('request');
+var Promise = require('bluebird');
+
+class Iterator {
+    constructor(adapter, req, res) {
+        this.points = [];
+        this.done = false;
```

Comment:
This constructor does a ludicrous amount of work, let's cut it off here at line 13 and move the rest into a `start()` method.